streams/src/main/java/org/apache/kafka/streams/processor/internals/StateDirectory.java

```diff
@@ -301,6 +301,22 @@ public synchronized void clean() {
             );
             throw new StreamsException(exception);
         }
+
+        try {
+            if (hasPersistentStores && stateDir.exists() && !stateDir.delete()) {
```

Comment:
```suggestion
            if (stateDir.exists() && !stateDir.delete()) {
```

Do we need to check `hasPersistentStores` here? It seems sufficient just to check if the directory exists.

Reply:
Exactly. But I thought keeping symmetry with the Consturctor is better.

```
if (this.hasPersistentStores && !stateDir.exists() && !stateDir.mkdir()) {
    throw new ProcessorStateException(
        String.format("state directory [%s] doesn't exist and couldn't be created", stateDir.getPath()));
}
```